test/cpp/api/modules.cpp

```diff
@@ -108,6 +108,20 @@ TEST_F(ModulesTest, MaxPool1d) {
   ASSERT_EQ(y.sizes(), torch::IntArrayRef({1, 1, 2}));
 }
 
+TEST_F(ModulesTest, MaxPool1dReturnIndices) {
+  MaxPool1d model(MaxPool1dOptions(3).stride(2));
```

Comment:
Hmm... Crazy idea, but instead of having a separate `forward_with_indices` method, would using templates help here?
So:
`MaxPool1d</*return_indices=*/true>` has a different `forward` function for each...

Reply:
I feel that it's a bit harder to read because it will introduce another way to specify an "option" for `MaxPool1d` besides using `MaxPool1dOptions`